addons/analytic/tests/test_analytic_account.py

```diff
@@ -275,3 +275,73 @@ def test_change_plan_no_conflict(self):
             plan_1_col: False,
             plan_2_col: self.analytic_account_1.id,
         }])
+
+    def test_change_parent_plan(self):
+        """Changing the parent of a plan updates account columns of the analytic lines."""
+        plan_temp = self.env['account.analytic.plan'].create({
+            'name': 'Temp Plan',
+            'default_applicability': 'optional',
+        })
+        analytic_account_temp = self.env['account.analytic.account'].create({
+            'name': 'Temp Account',
+            'plan_id': plan_temp.id,
+        })
+        plan_temp_col = plan_temp._column_name()
+        line = self.env['account.analytic.line'].create({
+            'name': 'test',
+            plan_temp_col: analytic_account_temp.id,
+        })
+
+        plan_temp.parent_id = self.analytic_plan_2
+        self.assertRecordValues(line, [{
+           self.analytic_plan_2._column_name(): analytic_account_temp.id,
+        }])
+        self.assertNotIn(plan_temp_col, line)
+
+    def test_change_parent_plan_conflicts(self):
+        """Test cases where some lines already have values set for the parent plan."""
+        plan_temp = self.env['account.analytic.plan'].create({
+            'name': 'Temp Plan',
+            'default_applicability': 'optional',
+        })
+        analytic_account_temp = self.env['account.analytic.account'].create({
+            'name': 'Temp Account',
+            'plan_id': plan_temp.id,
+        })
```

Comment:
You can just use `analytic_plan_1` and `analytic_plan_2` with `analytic_account_1`